doc/api/packages.md

```diff
@@ -612,17 +612,23 @@ The following condition definitions are currently endorsed by Node.js:
 * `"browser"` - any environment which implements a standard subset of global
   browser APIs available from JavaScript in web browsers, including the DOM
   APIs.
+* `"deno"` - indicates a variation for the Deno platform.
 * `"development"` - can be used to define a development-only environment
   entry point. _Must always be mutually exclusive with `"production"`._
 * `"production"` - can be used to define a production environment entry
   point. _Must always be mutually exclusive with `"development"`._
+* `"source"` - indicates the original .js source file without minification
+  or bundling optimizations, useful for development and debugging workflows.
+* `"types"` - can be used by typing systems to resolve the typing file for
```

Comment:
Oh, this definitely needs to call out that `types` needs to come _before_ other conditions that will be active like `import`, `require`, or `default` - otherwise it won't be matched. We're seeing _extreme_ confusion about this in the community. It's really bad.

Potentially also worth noting is that TS supports conditions like `"types@>=4.3"` so you can semver-limit your type resolutions.

Reply:
@weswigham is there a reference for the types versioning that we can link to here?

Reply:
I've noted that `"types"` should always come first. We do generally note that conditions should be from most specific to least specific. Ways to better clarify this are welcome. Perhaps a diagram that clearly demonstrates with an example that it's just like a set of if conditions would clearly get this across....

Reply:
We could also list the conditions in order of precedence here in this document, rather than putting `"types"` last.

Reply:
Good point - I've ordered all condition lists from most specific to least specific and added another note to each list about this.